host-controller/src/main/java/org/jboss/as/domain/controller/transformers/ServerGroupTransformers.java

```diff
@@ -47,9 +47,16 @@ static ChainedTransformationDescriptionBuilder buildTransformerChain() {
         //////////////////////////////////
         //The EAP/AS 7.x chains
 
-        // module-options was introduced in 13, WildFly 20
-        ResourceTransformationDescriptionBuilder currentTo13 = createBuilderFromCurrent(chainedBuilder, KernelAPIVersion.VERSION_13_0);
-        JvmTransformers.registerTransformers13_AndBelow(currentTo13);
+        // graceful-startup attribute was introduced in Kernel 16 (Wildfly 23)
+        ResourceTransformationDescriptionBuilder currentTo15 = createBuilderFromCurrent(chainedBuilder, KernelAPIVersion.VERSION_15_0)
+                .getAttributeBuilder()
+                .setDiscard(DiscardAttributeChecker.DEFAULT_VALUE, ServerGroupResourceDefinition.GRACEFUL_STARTUP)
+                .addRejectCheck(RejectAttributeChecker.DEFINED, ServerGroupResourceDefinition.GRACEFUL_STARTUP)
+                .end();
+
+        // module-options was introduced in 14, WildFly 20
```

Comment:
This says 14, but the code says 15.

I'm pretty sure that's because the 14 means WildFly Core 14, which used Kernel API version 15. This stuff is confusing and some day someone is going to mess up and we won't notice.

Can you change this (and others in this file) something like:

'blah blah Kernel X (WildFly Core Y, WildFly Z)'